git-ftp

```diff
@@ -1340,7 +1340,15 @@ check_remote_access() {
 	set_default_curl_options
 	CURL_ARGS+=(--ftp-create-dirs)
 	CURL_ARGS+=("$REMOTE_BASE_URL/$REMOTE_PATH")
-	curl "${CURL_ARGS[@]}" > /dev/null
+	if [ "$REMOTE_PROTOCOL" == "sftp" ]; then
```

Comment:
I think you can simplify this:

```sh
	curl "${CURL_ARGS[@]}" > /dev/null
	if [ "$REMOTE_PROTOCOL" == "sftp" -a $? -eq 78 ]; then
		write_log "Create $REMOTE_PATH"
		curl "${CURL_ARGS[@]}" -Q "MKDIR $REMOTE_PATH" > /dev/null
	fi
```

Reply:
I tried this before I came up with my current solution, but this does not work. It would be so much better.

The problem is the call of `check_exit_status` after the `if else`. It uses `$?` (the exit code of the last command). And if you do it like you suggested (and the protocol is not sftp), the last command is the if condition, so the `check_exit_status` would not work as expected. (For example if you do the curl with ftp but invalid credentials, the if part would be skipped but the check returns `true` so the program continues.)

An other idea I had was to cache the exit status of curl in a local variable, but then I can’t use `check_exit_status` either (can’t pass parameter) and must do the check on my own.

Reply:
You are right. Great work.

Reply:
Oh. As I noticed now, this same issue occurs inside `if [ "$REMOTE_PROTOCOL" == "sftp" ]; then`. If the curl with sftp return something else than `78` it runs into `check_exit_status` with the result of the inner if condition XD Since sftp is not covered by the automated tests and I run not into this on my custom testing, I didn't realize it until now.

So maybe the solution with the “cache” would be the better one. I’ll implement this (similar as here) and commit it. Thank’s for the hint to rethink it.
https://github.com/git-ftp/git-ftp/blob/ffd3e10c37f22761a06a92641ee9561b97eb98e0/git-ftp#L794-L805